Le lithium fix and le precision bucket

badgermemelord opened the pull request:
fixes lithium (not fully, theres lasaga (not) saving bug still)

badgermemelord committed:
test

badgermemelord committed:
Cleanup

badgermemelord committed:
Precision Bucket v1

badgermemelord committed:
Add lang and texture

badgermemelord committed:
Add tooltip

badgermemelord committed:
cleanup

badgermemelord committed:
remove unused dependency

badgermemelord committed:
cleanup

badgermemelord committed:
Conditional Item texture

badgermemelord committed:
add fill level bar

badgermemelord committed:
wip

badgermemelord committed:
Precise volumes for game logic interaction working

badgermemelord committed:
Cleanup

badgermemelord committed:
Cleanup pt2

Also, change bucket texture logic

ewoudje commented on src/client/java/com/wetwater/washwater/WaterModClient.java:
```diff
@@ -1,10 +1,40 @@
 	}
+
+	public static void registerItemProperties() {
+		// For versions before 1.21, replace 'Identifier.ofVanilla' with 'new Identifier'.
```
Remove comments

ewoudje commented on src/main/java/com/wetwater/washwater/FluidSection.java:
```diff
@@ -33,7 +33,7 @@ public class FluidSection implements ExtraSectionStorage {
     private boolean isSavedDirty = false;
     private int amountOfWaters = 0;
-    private final LevelChunk chunk;
+    public final LevelChunk chunk;
```
Don't.

ewoudje commented on src/main/java/com/wetwater/washwater/item/BucketMechanics.java:
```diff
@@ -39,6 +42,51 @@ public static boolean creativePipettePlace(Level level, BlockPos pos, ItemStack
             }
         return true;
```
Move all code in this class to thier relevant item classes, it doesn't make much sense to put it here instead. I don't think its ever called from somewhere else right?

ewoudje commented on src/main/java/com/wetwater/washwater/item/FluidPipetteItem.java:
```diff
@@ -20,19 +20,16 @@ public InteractionResult useOn(UseOnContext useOnContext) {
-                BucketMechanics.creativePipettePickup(level, targetPos, itemStack, player);
-            }
-
+        if (!player.isCrouching()) {
```
Remove the logs

ewoudje commented on src/main/java/com/wetwater/washwater/item/FluidPipetteItem.java:
```diff
@@ -20,19 +20,16 @@ public InteractionResult useOn(UseOnContext useOnContext) {
-
+        if (!player.isCrouching()) {
+            System.out.println("Placed Water with Pipette");
+            BucketMechanics.creativePipettePlace(level, targetPos, itemStack, player);
```
So instead of calling BucketMechanics just put that method here. KEEP it in a seperate method

badgermemelord replied:
amongus

ewoudje commented on src/main/java/com/wetwater/washwater/item/PrecisionBucketItem.java:
```diff
@@ -0,0 +1,77 @@
+        }
+        if (player != null) {
+            if (!player.isCrouching()) {
+                BucketMechanics.precisionBucketPlace(level, targetPos, itemStack, player);
```
Move the method to this class

ewoudje commented on src/main/java/com/wetwater/washwater/mixin/FluidStateMixin.java:
```diff
@@ -0,0 +1,51 @@
+        ww$volume = volume;
+    }
+
+/*    @Inject(at = @At("HEAD"), method = "getHeight", cancellable = true)
```
remove comment

ewoudje commented on src/main/java/com/wetwater/washwater/mixin/MixinLevelChunkSection.java:
```diff
@@ -36,11 +41,64 @@ public void setBlockState(int x, int y, int z, BlockState state, boolean lock, C
+    public void getFluidState(int x, int y, int z, CallbackInfoReturnable<FluidState> cir) {
+        if (fluidSection != null && cir.getReturnValue().isEmpty()) {
+            WashWaterFluidState waterState = ((WashWaterFluidState)(Object)WaterInfo.getWaterState(fluidSection.getWaterVolume(x, y, z)));
+            waterState.ww$setVolume(fluidSection.getWaterVolume(x, y, z));
```
move the ww$setVolume into WaterInfo.getWaterState

ewoudje commented on src/main/java/com/wetwater/washwater/mixin/MixinLevelChunkSection.java:
```diff
@@ -36,11 +41,64 @@ public void setBlockState(int x, int y, int z, BlockState state, boolean lock, C
+    }
+    @Inject(at = @At("RETURN"), method = "maybeHas", cancellable = true)
+    public void maybeHas(Predicate<BlockState> predicate, CallbackInfoReturnable<Boolean> cir) {
+        //System.out.println("Test1");
```
comments

ewoudje commented on src/main/java/com/wetwater/washwater/mixin/MixinLevelChunkSection.java:
```diff
@@ -36,11 +41,64 @@ public void setBlockState(int x, int y, int z, BlockState state, boolean lock, C
+     * @reason
+     */
+/*    @Overwrite
+    public boolean maybeHas(Predicate<BlockState> predicate) {
```
comments

ewoudje reviewed (changes requested):
DENIED

badgermemelord committed:
Now compatible with NOT-Lithium

game doesnt crash when lithium isnt present
mixin plugin
needs investigation regarding whether kotlin issues apply

badgermemelord committed:
wip

badgermemelord committed:
Good enough of a solution

Fixes minY crash

badgermemelord committed:
Fix chunk edge water rendering

badgermemelord committed:
[Rendering Fix] Fix brightness

someone forgor to fully implement sodium code for quad brightness

badgermemelord committed:
debugging

badgermemelord committed:
PR fix

badgermemelord committed:
Replace dollar signs with appropriate ones
